indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/WorkerTaskRunnerConfig.java

```diff
@@ -29,14 +34,17 @@ public class WorkerTaskRunnerConfig
   @JsonProperty
   private double parallelIndexTaskSlotRatio = 1;
 
+  @JsonProperty
+  private Map<String, @Min(value = 0, message = "Task slot ratio for must be at least 0.") @Max(value = 1, message = "Task slot ratio cannot be greater than 1") Double> taskSlotRatiosPerWorker = new HashMap<>();
```

Comment:
I'm not sure that adding the constructor would be the best solution here as this is the parent for a couple other classes, and it will require children refactoring as well (including default values specifying), min max annotations seems to be a better approach, if it'll go together with JsonConfigurator update, otherwise I'd better stick with CustomDeserializer validation approach

Reply:
Hmm, fair point. In that case, let's just skip the validation altogether as there is no validation currently happening for `parallelIndexTaskSlotRatio` either. We can add it when needed later.